### DISCUSSION TITLE
Release 1.2.0 Beta 8

### DISCUSSION THREAD

**Discussion by apohl79** [apohl79/audiogridder#942] [General]
## PLUGIN
 * Fixed service receiver reachable issue:
    The server list was updated before checking the reachability and cleaned up afterwards. This could lead to the plugin triggering a reconnect. This issue materialized on Windows with ipv4 and ipv6 enabled.

## SERVER 
 * Optimized local mode
    This change avoids flickering when opening a plugin UI by rendering it at the target position initially to avoid moving it and by waiting for editor creation in the worker thread.

<hr /><em>This discussion was created from the release <a href='https://github.com/apohl79/audiogridder/releases/tag/release_1_2_0_beta8'>Release 1.2.0 Beta 8</a>.</em>

**Comment by Pabloscotty** [comment]
Thanks again :)

**Comment by djschultz123** [comment]
Thanks! This seems, on first load, to have fixed the issues I had with plugin location AND the menu of the loaded plugin not being usable. I'll report back if I have issues with other plugins I load in - so far, so good! 

EDIT: The server can't seem to find/display the plugin Kush Blyss (yes, the plugin works great and is installed in the usual VST3 location on my drive).

**Reply by apohl79** [reply]
Is it black listed in the AG plugin manager? If yes, can you remove it from the black list and try a rescan?

**Reply by djschultz123** [reply]
It wasn't in the blacklist - the log files show it was scanned successfully and added to FX:EQ, but it doesn't actually appear in there (or in the search). I've cleared the cache and rescanned, as well. Still no luck.

Here's the content of one of the log files:

[16:23:35.494|app|21850995778] scan mode: format=VST3 id=C:\Program Files\Common Files\VST3\Blyss.vst3
[16:23:35.494|server|7ff6578ac910] scanning id=C:\Program Files\Common Files\VST3\Blyss.vst3 fmt=VST3
[16:23:36.336|server|7ff6578ac910] adding plugin description:
[16:23:36.336|server|7ff6578ac910]   name            = Blyss (Blyss)
[16:23:36.336|server|7ff6578ac910]   uniqueId        = b334a4d
[16:23:36.336|server|7ff6578ac910]   deprecatedUid   = 10d4b4c
[16:23:36.336|server|7ff6578ac910]   id string       = VST3-Blyss-10d4b4c
[16:23:36.336|server|7ff6578ac910]   manufacturer    = Kush Audio
[16:23:36.336|server|7ff6578ac910]   category        = Fx|EQ
[16:23:36.336|server|7ff6578ac910]   shell           = 0
[16:23:36.336|server|7ff6578ac910]   instrument      = 0
[16:23:36.336|server|7ff6578ac910]   input channels  = 0
[16:23:36.336|server|7ff6578ac910]   output channels = 0
[16:23:36.341|app|21850995778] ...success

**Reply by apohl79** [reply]
And you see it in the plugin manager right? My assumption is, that the following

> [16:23:36.336|server|7ff6578ac910] input channels = 0
> [16:23:36.336|server|7ff6578ac910] output channels = 0

causes the server to filter it out, as the plugin does not report any channels during the scan. Try deactivating the server filter in the plugin under "Plugin Menu".

**Reply by djschultz123** [reply]
Success! Deactivating the server filter revealed the Blyss plugin. Thanks for your time to help me figure that out (and for the product in the first place!).

**Comment by dmuckala** [comment]
Just tested the instrument version in local mode with Logic Pro/Intel Monterey 12.01 and getting buffer distortion.  Previous version 1.2 beta 6 worked fine in Logic however.  
I messed with logic's "process buffer range" to see if that would fix it but no luck.

**Reply by JamesXXV** [reply]
^ Same issue here on M1 Pro Monterey 12.2.1, Logic Pro 10.7.3

**Comment by dmuckala** [comment]
Ok fyi I found one combo that works.  'Buffer size' (which is the interface or main buffer size) of 512 and 'process buffer range' (which is internal buffer of Logic) of "small".  This is a lot more restrictive than the previous version.   I was usually able to run 256 or 128 with a "medium" buffer and there were no artifacts.

**Reply by apohl79** [reply]
Is this for an instruments or fx channel?

**Reply by dmuckala** [reply]
instrument channel

**Reply by apohl79** [reply]
I can't reproduce this. I'm not on monterey though. Tested on big sur with logic 10.7.0, kontakt player and AS local mode. All settings from 32 samples I/O with small, medium and large "Process Buffer Range" settings to large I/O buffers. I also changed the AG blocks from 0 to 8 and things in between. No audio issues whatsoever.

If you get a chance, please provide plugin and server logs and open a help thread.

**Comment by emack999** [comment]
> * Fixed service receiver reachable issue:
>   The server list was updated before checking the reachability and cleaned up afterwards. This could lead to the plugin triggering a reconnect. This issue materialized on Windows with ipv4 and ipv6 enabled.

I think this is fixed now---only did a quick test on it but IPv4 and IPv6 running together didn't cause any issues like in beta7.

**Comment by fojerhar** [comment]
This update works here flawlessly
Incredible loading times, really responsive
And there are no any issues with missing plugins anymore
thank you, amazing job
(AMD hackintosh, monterey, PT21.04, local mode)

**Reply by emack999** [reply]
I agree---this is by far the most stable and responsive version of AG to date.

**Reply by apohl79** [reply]
thanks for the feedback, very much appreciated!

**Reply by fojerhar** [reply]
Has some crashes today, while I was mixing.
Nothing special happened, PT started beachballing, and froze
logs:
https://www.dropbox.com/sh/jhedjliskc1ezek/AACsyvUgV6vA8sd5MuZVZPJia?dl=0

hope this can help
Thank you, still. amazing
